lib/board.rb

```diff
@@ -0,0 +1,60 @@
+class Board   #attributes externally visible
+  attr_accessor :cells
+
+  #resets to empty array
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display(cells= @cells)
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  #takes in index and returns value
+  def position(location)
+    @cells[location.to_i - 1]
+  end
+
+  #returns true if board entirely occupied with X or O
+  def full?(cells = @cells)
+    cells.all?{|taken| taken == "X" || taken == "O"}
+  end
+
+  #returns turn count based on how many positions in cell array are filled
+  def turn_count(cells = @cells)
+  turn = 0
+    cells.each do |taken|
+      if taken == "X" || taken == "O"
+    turn += 1
+    end
+    end
+    turn
+  end
+
+  #takes in index and returns true if value is X or O
+  def taken?(cells = @cells, location)
+    if @cells[location.to_i - 1] == "X" || @cells[location.to_i - 1] == "O"
+      return true
+    else
+      return false
+    end
+  end
```

Comment:
Could we simplify this down to one line using Ruby's implicit return?  Also, since we have access to the board, do we need to pass `cells`?  It should work the same way with just your condition:
```
def taken?(location)
  @cells[location.to_i - 1] == "X" || @cells[location.to_i - 1] == "O"
end
```